src/main/java/org/spongepowered/api/block/BlockType.java

```diff
@@ -121,4 +122,18 @@
      */
     float getEmittedLight();
 
+    /**
+     * Gets if the block is able to be gathered using the silk touch
+     * enchantment.
+     *
+     * @return If the block can be harvested with silk touch
+     */
+    boolean canHarvestWithSilk();
```

Comment:
My idea was to have `Enchantments.SILK_TOUCH` a special interface as defined below:
```java
public interface EnchantmentSilkTouch extends Enchantment {

    /**
     * Register a block that can be harvested with silk touch.
     *
     * @param block The block to register
     */
    void registerBlock(BlockType block);

    /**
     * Unregister a block so it can no longer be harvested with silk touch.
     *
     * @param block The block to unregister
     */
    void unregisterBlock(BlockType block);

    /**
     * Gets whether the block can be harvested with silk touch.
     *
     * @param block The block to test for
     * @return Whether it can be harvested
     */
    boolean canHarvest(BlockType block);
}
```
The vanilla blocks are added on start-up, but plugins now have the ability to modify what can and can't be harvested with silk touch.

For the implementation, `canSilkHarvest` would need to be modified to return `Enchantments.SILK_TOUCH.canHarvest(this)` which is straightforward.